app/utils/Validation.scala

```diff
@@ -0,0 +1,22 @@
+package utils
+
+import org.joda.time.YearMonth
+import org.joda.time.format.DateTimeFormat
+
+import scala.util.Try
+
+/**
+ * Created by coolit on 16/03/2018.
+ */
+object Validation {
+  private val minKeyLength: Int = 5
```

Comment:
We probably need to double-check the requirements for key length.  I was told the minimum was 4, with no upper limit ...

Reply:
Yes I agree. It should be id.length >= minKeyLength, where minKeyLength is 4.